test/read-line.l

```diff
@@ -2,18 +2,34 @@
 
 (init-unit-test)
 
+#|
 (defun test-read-line
   (max-char)
   (length (read-line
            (make-string-input-stream
             (apply #'concatenate string
                    (make-list max-char :initial-element "a")))))
   )
+|#
+
+(defun test-read-line
+  (max-char)
+  (let ((sm (make-string-output-stream)))
+    (dotimes (i max-char)
+      (format sm "~C" #\a))
+    (length (read-line
+             (make-string-input-stream (get-output-stream-string sm))))
+    )
+  )
+
+
 
 (deftest test-read-line-all
   (format *error-output* " read-line test~%")
   (assert (= (test-read-line 8191) 8191) "read-line with ~A char" 8191)
-  ;; (assert (= (test-read-line 8192) 8192) "read-line with ~A char" 8192) ;; tempolarily comment out
+  (assert (= (test-read-line 8192) 8192) "read-line with ~A char" 8192) ;; tempolarily comment out
+  (assert (= (test-read-line (- 1000000 8192)) (- 1000000 8192)) "read-line with ~A char" (- 1000000 8192))
+  (assert (/= (test-read-line 1000000) 1000000) "read-line with ~A char" 1000000) ;; may print 'line length exceeds MAX_READLINE_LNEGHT(1000000)'
```

Comment:
@YoheiKakiuchi 

The test passed with `line length exceeds MAX_READLINE_LNEGHT(1000000)'`, did we change something?
```
start testing [test-read-line-all]
```
start testing [test-read-line-all]
 read-line test
[ERROR] test (/= (test-read-line 1000000) 1000000) failed ... (read-line with 1000000 char  read-line test
ng [test-read-line-all]
d on p51s(Wed Aug 22 09:23:07 JST 2018)^[0m).
TEST-NAME: test-read-gensym
  now testing...
start testing [test-read-gensym]
```

Reply:
limitation of read-line was removed.
Then, it's wrong.